src/main/java/com/hopper/cloud/airlines/HopperClient.java

```diff
@@ -175,14 +230,7 @@ public boolean processCfarPayment(String sessionId, String contractId, ProcessCf
             tokenizationRequest.getPaymentMethod().setCreditCard(new CreditCard());
             tokenizationRequest.getPaymentMethod().getCreditCard().setFirstName(processCfarPaymentRequest.getFirstName());
             tokenizationRequest.getPaymentMethod().getCreditCard().setLastName(processCfarPaymentRequest.getLastName());
-
-            // Map test credit cards to an accepted one on staging environment
-            if (cfarApi.getApiClient().getBasePath().contains("staging") && Arrays.asList("371449635398431", "5454545454545454").contains(processCfarPaymentRequest.getNumber())) {
-                tokenizationRequest.getPaymentMethod().getCreditCard().setNumber("4111111111111111");
-            } else {
-                tokenizationRequest.getPaymentMethod().getCreditCard().setNumber(processCfarPaymentRequest.getNumber());
-            }
-
+            tokenizationRequest.getPaymentMethod().getCreditCard().setNumber(prepareCreditCardNumberForSpreedly(processCfarPaymentRequest.getNumber()));
```

Comment:
FMI: would it be cleaner to share the object reference? Seems like it would be much easier to read.

```java
CreditCard creditCard = new CreditCard();
creditCard.setFirstName(...)
creditCard.setLastName(...)

tokenizationRequest.getPaymentMethod().setCreditCard(creditCard);
```

Even if we don't do it for this entire method, we probably should at least do it for the credit card.

Reply:
Correct ! This is historic (and existing) code. I just updated this method by adding the call to the new utility method `prepareCreditCardNumberForSpreedly`
I'll adjust that